Agents/DistrictHeatingNetworkAgent/DistrictHeatNetworkAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatNetworkInputAgent.java

```diff
@@ -0,0 +1,686 @@
+                new UpdateBuilder()
+                        .addInsert(NodeFactory.createURI(TSIRI), NodeFactory.createURI(RDF_TYPE), NodeFactory.createURI(OM_MEASURE))
+                        .addInsert(NodeFactory.createURI(TSIRI), NodeFactory.createURI(OM_HAS_UNIT), NodeFactory.createURI(KB + Unit))
+                        .addInsert(NodeFactory.createURI(KB + Unit), NodeFactory.createURI(RDF_TYPE), NodeFactory.createURI(OM_UNIT))
+                        .addInsert(NodeFactory.createURI(KB + Unit), NodeFactory.createURI(OM_SYMBOL), Unit);
+        UpdateRequest omHasValueTS_ur = omHasValueTS_ub.buildRequest();
+        AccessAgentCaller.updateStore("http://host.docker.internal:48888/ontoheatnet", omHasValueTS_ur.toString());
+    }
+
+   // Setter for the time series client.
+   // @param tsClient The time series client to use.
+    public void setTsClient(TimeSeriesClient<OffsetDateTime> tsClient) {
+        this.tsClient = tsClient;
+    }
+
+    // Update the TS database with new readings.
+    public void updateTSData(Map<String, List<?>> HeatingNetworkReadings) throws IllegalArgumentException {
+        if (!HeatingNetworkReadings.isEmpty()) {
+            List<TimeSeries<OffsetDateTime>> timeSeries;
+            try {
+                timeSeries = convertReadingsToTimeSeries(HeatingNetworkReadings);
+            } catch (NoSuchElementException e) {
+                throw new IllegalArgumentException("Readings can not be converted to proper time series!", e);
+            }
+            for (TimeSeries<OffsetDateTime> ts : timeSeries) {
+                if (!ts.getTimes().isEmpty()) {
+                    try {
+                        tsClient.addTimeSeriesData(ts);
+                    } catch (Exception e) {
+                        e.getMessage();
+                       throw new JPSRuntimeException("Cannot not add timeseries!");
+                    } finally {
+                        tsClient.disconnectRDB();
+                    }
+                }
+            }
+        } else {
+            throw new IllegalArgumentException("Readings can not be empty!");
+        }
+    }
+
+
+    // Converts the readings in form of maps to time series' using the mappings to IRI.
+    // @return A list of time series objects (one per mapping) that can be used with the time series client.
+    private List<TimeSeries<OffsetDateTime>> convertReadingsToTimeSeries(Map<String, List<?>> HeatingNetworkReadings)
+            throws NoSuchElementException, IOException {
+        List<OffsetDateTime> HeatingNetworkTimestamps = HeatingNetworkReadings.get(HeatNetworkInputAgent.timestampKey).stream()
+                .map(timestamp -> (convertStringToOffsetDateTime(timestamp.toString()))).collect(Collectors.toList());
+        List<TimeSeries<OffsetDateTime>> timeSeries = new ArrayList<>();
+        //String mappingFolder = "/Users/HXUE01/Documents/GitHub/TheWorldAvatar/Agents/DistrictHeatingNetworkAgent/config/mappings";
```

Comment:
Please remove this comment.

Reply:
I have done all the amendments according to the comments. Thank you very much for the feedback and please let me know any further changes that are required.